src/pages/Landing/index.tsx

```diff
@@ -300,35 +301,28 @@ const Link = styled(NativeLink)`
 
 export default function Landing() {
   const isDarkMode = useIsDarkMode()
-
   const cardsRef = useRef<HTMLDivElement>(null)
-
-  const [showContent, setShowContent] = useState(false)
   const selectedWallet = useAppSelector((state) => state.user.selectedWallet)
-  const navigate = useNavigate()
-  const location = useLocation()
-  const queryParams = parse(location.search, {
-    ignoreQueryPrefix: true,
-  })
+  const shouldDisableNFTRoutes = useDisableNFTRoutes()
+  const cards = useMemo(
+    () => MAIN_CARDS.filter((card) => !(shouldDisableNFTRoutes && card.to.startsWith('/nft'))),
+    [shouldDisableNFTRoutes]
+  )
 
   const [accountDrawerOpen] = useAccountDrawer()
+  const navigate = useNavigate()
   useEffect(() => {
-    if ((queryParams.intro || !selectedWallet) && !accountDrawerOpen) {
-      setShowContent(true)
-    } else {
-      setShowContent(false)
+    if (accountDrawerOpen) {
```

Comment:
I expected:
`if ((!queryParams.intro && selectedWallet) || accountDrawerOpen)`

Reply:
This is intentionally split. Handling both as navigations (ie `navigate` or `Redirect`) allows us to avoid having any state, and keeps the landing page "purer" by not having it mimic the /swap page.

`!queryParams.intro && selectedWallet` can be redirected immediately. They are state that exist on pageload, and so can be handled immediately to avoid even rendering the landing page.

`accountDrawerOpen` is dependent on user action after pageload, so it needs to be handled separately, in a `useEffect`. It also needs to be delayed so that the account drawer _can_ open before navigating, to avoid jank.